baybe/surrogates/utils.py

```diff
@@ -4,15 +4,20 @@
 
 from collections.abc import Callable
 from functools import wraps
-from typing import TYPE_CHECKING
-
-from baybe.scaler import DefaultScaler
-from baybe.searchspace import SearchSpace
+from typing import TYPE_CHECKING, Any
 
 if TYPE_CHECKING:
+    from botorch.posteriors import Posterior
     from torch import Tensor
 
     from baybe.surrogates.base import Surrogate
+    from baybe.surrogates.naive import MeanPredictionSurrogate
+
+
+_constant_target_model_store: dict[int, MeanPredictionSurrogate] = {}
```

Comment:
do you really prefer this solution over the non-slotted version attaching the mean-pred model to the instance?

Reply:
I can share my thoughts why I changed it and then you tell me if you agree or not:
* As also stated [here](https://github.com/emdgroup/baybe/pull/309#discussion_r1671668184), the slot disabling was out of sheer necessity due to the previous implementation, but not because I wanted it. I think having the decorator modify the attributes of the class is not ideal, because you lose the decoupling of the two components, and also stuff like auto-completion isn't fully supported there. And all in all, it caused quite some overhead in the decorator logic, making it much more difficult to understand.
* Which brings me to the second point: the new version is just much simpler, i.e. requires significantly less and clearer code. That alone is typically an indication that something improved and would be reason enough for me.
* And perhaps the most pragmatic reason: I (at least currently) see now drawbacks with this approach. We have a nice decoupling, and easy-to-follow logic. What else do we need?

Reply:
`reasonable` under the assumption that we need `slots`

Reply:
Well, "need" is perhaps exaggerated, but definitely desired. OK, so let's stick with it for now. This is basically all internal logic, so we can easily adjust if we ever need to change again / find an even better solution 👍🏼